server/src/main/java/org/elasticsearch/inference/InferenceService.java

```diff
@@ -36,40 +36,55 @@ default void init(Client client) {}
      * If the map contains unrecognized configuration option an
      * {@code ElasticsearchStatusException} is thrown.
      *
-     * @param modelId               Model Id
-     * @param taskType              The model task type
-     * @param config                Configuration options including the secrets
-     * @param parsedModelListener   A listener which will handle the resulting model or failure
+     * @param modelId             Model Id
+     * @param taskType            The model task type
+     * @param config              Configuration options including the secrets
+     * @param endpointVersion
+     * @param parsedModelListener A listener which will handle the resulting model or failure
      */
-    void parseRequestConfig(String modelId, TaskType taskType, Map<String, Object> config, ActionListener<Model> parsedModelListener);
+    void parseRequestConfig(
+        String modelId,
+        TaskType taskType,
+        Map<String, Object> config,
+        String endpointVersion,
```

Comment:
Why does `endpointVersion` need to be split from `config` here? I think it'll greatly reduce the code if we just pull it out from `config` when we actually need it (likely when building the ModelConfiguration). Then all we have to do is update the `getTaskSettings` function in `Model.java` and we'll be able to handle it without having to modify every service. In the future we'll likely want to build a parser that will handle this all anyways so we'll have to remove the code from each layer regardless.

Reply:
In this factorization we handle the identification of the endpointVersion in the TransportPutInferenceModelAction and the rewriting of the config in the PutInferenceModelAction. This way, we don't need to make changes to all of the services serialization code (we do make changes to all of the function signatures, but just to pass along this variable, not to the serialization code itself). This way, none of the service-specific serialization code gets changed. 

I don't really see how modifying `getTaskSettings` would work, as I think that happens after we've done all the serialization?